### PR TITLE
Add clang tidy review in Lint CI

### PR DESCRIPTION
**Pull request by jcwchen**
### Description

Add clang tidy review developed by ZedThree/clang-tidy-review in Lint CI.

### Motivation and Context

Closes https://github.com/onnx/onnx/issues/4459.

### CODE REVIEW

File: onnx/shape_inference/implementation.cc
Comment (jcwchen) [inline onnx/shape_inference/implementation.cc:1010]: FYI: Just for experiment. Will remove it before PR is merged.

File: .github/workflows/lint.yaml
Comment (jcwchen) [reply .github/workflows/lint.yaml:48]: It's a valid concern. In that case I won't let clang-tidy check fail any CI, but instead have clang-tidy review every PRs and comment. The author can decide whether to take advice or not on their own.

**Comment by justinchuby** [comment]
image

looking good

**Review by jcwchen — COMMENTED** [review]
Please note that we can only test this new CI after the CI file is included in the main branch. At least, I have tested this CI in my forked repo and it seems working.

**Comment by justinchuby** [comment]
Was there any permission we need to explicitly enable in the yaml? Since you have write access I thought it should work for you

**Comment by jcwchen** [comment]
> Was there any permission we need to explicitly enable in the yaml? Since you have write access I thought it should work for you

Good question. See the document from the author: https://github.com/marketplace/actions/clang-tidy-review#usage-in-fork-environments-split-workflow. It seems the PR coming from the forked one cannot post a comment before merging into the main branch. My experiment also proves that.

**Review by liqunfu — APPROVED** [review]

**Pull request merged**
